src/v/raft/buffered_protocol.cc

```diff
@@ -254,10 +259,23 @@ append_entries_queue::append_entries_queue(
         _current_max_inflight_requests = new_value;
     });
     setup_internal_metrics();
+    vlog(_logger.info, "Creating new append entries queue");
     // start dispatch loop
-    ssx::repeat_until_gate_closed(
-      _gate,
-      [this, gate_holder = std::move(gate_holder)] { return dispatch_loop(); });
+    ssx::spawn_with_gate(
+      _gate, [this, gate_holder = std::move(gate_holder)]() mutable {
+          return ss::with_scheduling_group(
+            _sg, [this, gate_holder = std::move(gate_holder)] {
+                return ss::do_until(
+                  [this] { return _gate.is_closed(); },
+                  [this]() {
+                      return dispatch_loop().handle_exception(
```

Comment:
Why has the exception handling changed? Is this trying to do what `repeat_until_gate_closed` was doing before?

Why not just wrap the original line 258 with `with_scheduling_group`?

... or coroutinize.